src/vsphere_cpi/spec/unit/cloud/vsphere/nsxt_provider_spec.rb

```diff
@@ -219,7 +220,7 @@
 
     before do
       allow_any_instance_of(VSphereCloud::NSXTProvider).to receive(:grouping_obj_svc).and_return(grouping_obj_svc)
-      allow(grouping_obj_svc).to receive_message_chain(:list_ns_groups, :results).and_return([nsgroup_1, nsgroup_2])
+      allow(nsxt_provider).to receive(:retrieve_all_ns_groups_with_pagination).and_return([nsgroup_1, nsgroup_2])
```

Comment:
Use `expect` instead of `allow`

Expect makes unit test more water-tight by counting the number of times it is called. If your program is non-deterministic about number of calls, `allow` is fine. Otherwise, use `expect`

Reply:
Done